__tests__/workers/postUpdated.ts

```diff
@@ -0,0 +1,174 @@
+  expect(post.metadataChangedAt).toEqual(new Date('2020-01-05T12:00:00.000Z'));
+});
+
+it(`should not update if the post object doesn't have ID`, async () => {
+  await con
+    .getRepository(ArticlePost)
+    .update({ id: 'p1' }, { origin: PostOrigin.CommunityPicks });
+  await expectSuccessfulBackground(worker, {
+    updated_at: new Date('01-05-1990 12:00:00'),
+  });
+  const post = await con.getRepository(ArticlePost).findOneBy({ id: 'p1' });
+  expect(post.metadataChangedAt).toEqual(new Date('2020-01-05T12:00:00.000Z'));
+});
+
+it('should update the post and keep it invisible if title is missing', async () => {
+  await expectSuccessfulBackground(worker, {
+    id: 'p1',
+    updated_at: new Date('01-05-2023 12:00:00'),
+  });
+  const post = await con.getRepository(ArticlePost).findOneBy({ id: 'p1' });
+  expect(post.metadataChangedAt).toEqual(new Date('2023-01-05T12:00:00.000Z'));
+  expect(post.visible).toEqual(false);
+});
+
+it('should update the post and make it visible if title is available', async () => {
+  await expectSuccessfulBackground(worker, {
+    id: 'p1',
+    updated_at: new Date('01-05-2023 12:00:00'),
+    title: 'test',
+  });
+  const post = await con.getRepository(ArticlePost).findOneBy({ id: 'p1' });
+  expect(post.metadataChangedAt).toEqual(new Date('2023-01-05T12:00:00.000Z'));
+  expect(post.visible).toEqual(true);
+  expect(post.visibleAt).toEqual(new Date('2023-01-05T12:00:00.000Z'));
+  expect(post.title).toEqual('test');
+});
+
+it('should update the post related shared post to visible', async () => {
+  await createSharedPost();
+  await expectSuccessfulBackground(worker, {
+    id: 'p1',
+    updated_at: new Date('01-05-2023 12:00:00'),
+    title: 'test',
+  });
+  const post = await con.getRepository(ArticlePost).findOneBy({ id: 'p1' });
+  expect(post.metadataChangedAt).toEqual(new Date('2023-01-05T12:00:00.000Z'));
+  expect(post.visible).toEqual(true);
+  expect(post.visibleAt).toEqual(new Date('2023-01-05T12:00:00.000Z'));
+  expect(post.title).toEqual('test');
+  const sharedPost = await con
+    .getRepository(SharePost)
+    .findOneBy({ id: 'sp1' });
+  expect(sharedPost.visible).toEqual(true);
+  expect(sharedPost.visibleAt).toEqual(new Date('2023-01-05T12:00:00.000Z'));
+});
+
+it('should save a new post with the relevant keywords', async () => {
+  await createDefaultKeywords();
+  await expectSuccessfulBackground(worker, {
+    id: 'p1',
+    updated_at: new Date('01-05-2023 12:00:00'),
+    title: 'test',
+    keywords: ['alpine', 'a-b-testing', 'mongodb'],
+  });
+  const post = await con.getRepository(ArticlePost).findOneBy({ id: 'p1' });
+  expect(post.metadataChangedAt).toEqual(new Date('2023-01-05T12:00:00.000Z'));
+  expect(post.visible).toEqual(true);
+  expect(post.visibleAt).toEqual(new Date('2023-01-05T12:00:00.000Z'));
+  expect(post.title).toEqual('test');
+  const keywords = await con.getRepository(Keyword).find({
```

Comment:
We are not checking completely the keywords here, let's add a more thorough check on `tagsStr`